submission/distributor.go

```diff
@@ -68,12 +74,68 @@ func (d *Distributor) Run(ctx context.Context) {
 	}
 }
 
-// refreshRoots requests roots from Logs and updates local copy if necessary.
+// refreshRoots requests roots from Logs and updates local copy.
 func (d *Distributor) refreshRoots(ctx context.Context) {
-	// TODO(Mercurrent) add implementation.
+	type RootsResult struct {
+		LogURL string
+		Roots  []*x509.Certificate
+		Err    error
+	}
+	ch := make(chan RootsResult, len(d.logClients))
+
+	rctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Why do make `rctx` inherit from `context.Background()` instead of `ctx`?

Reply:
oops